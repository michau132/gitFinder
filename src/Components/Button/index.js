/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './style.css';

const Button = ({
  type, className, onClick, text, disabled,
}) => (
  <button
    disabled={disabled}
    type={type}
    className={classNames(className, styles.button, { [styles.disabledBtn]: disabled })}
    onClick={onClick}
  >
    {text}
  </button>
);

Button.defaultProps = {
  className: null,
  onClick: null,
  disabled: false,
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
};

export default Button;
