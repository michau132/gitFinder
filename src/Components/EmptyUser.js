import React from 'react';
import { Typography } from '@material-ui/core';

const EmptyUser = () => (
  <Typography variant="h5">
    Here will be a profile from Github. Enter above a Github login to see more.
  </Typography>
);

export default EmptyUser;
