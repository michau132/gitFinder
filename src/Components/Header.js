import React from 'react';
import PropTypes from 'prop-types';
import {
  TextField,
  Button,
  Tooltip,
  Grid,
} from '@material-ui/core';
import { Search } from '@material-ui/icons';
import styled from 'styled-components';


const StyledTooltip = styled(props => (
  <Tooltip classes={{ popper: props.className, tooltip: 'asd' }} {...props} />
))`
  & .asd {
    font-size: 15px;
    max-width: 600px;
  }
`;

const StyledTextField = styled(TextField)`
  width: calc(100% - 115px);
`;

const StyledHeader = styled.header`
  background-color: #f5f5f5;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
  margin-bottom: 15px;
`;

const Header = ({
  onFormSubmit, inputValue, updateInputValue, errorInput,
}) => (
  <StyledHeader>
    <form
      onSubmit={onFormSubmit}
    >
      <Grid container justify="space-between">
        <StyledTooltip
          open={errorInput}
          title="Invalid github login. In login you can&apos;t have special signs like blank space, @, #, etc."
          disableFocusListener
          disableHoverListener
          disableTouchListener
        >
          <StyledTextField
            type="search"
            error={errorInput}
            onChange={updateInputValue}
            value={inputValue}
            label="Github login"
            variant="outlined"
            color="inherit"
          />
        </StyledTooltip>
        <Button type="submit" variant="contained" color="primary">
            Search
          <Search />
        </Button>
      </Grid>
    </form>
  </StyledHeader>
);

Header.defaultProps = {
  errorInput: null,
};

Header.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
  updateInputValue: PropTypes.func.isRequired,
  inputValue: PropTypes.string.isRequired,
  errorInput: PropTypes.bool,
};

export default Header;
