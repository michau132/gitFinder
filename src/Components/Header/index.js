import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';

const Header = ({ onFormSubmit, historyPush }) => {
  let input;
  return (
    <form
      onSubmit={(e) => {
        e.preventDefault();
        onFormSubmit(input.value);
        historyPush(`/${input.value}`);
      }}
      className={styles.form}
    >
      <input type="text" className={styles.formInput} ref={node => input = node} placeholder="github login" />
      <button type="submit" className={styles.formButton}>Search</button>
    </form>
  );
};

Header.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
  historyPush: PropTypes.func.isRequired,
};

export default Header;
