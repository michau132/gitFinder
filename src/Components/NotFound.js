import React from 'react';
import { Typography } from '@material-ui/core';

const NotFound = () => (
  <Typography variant="h5">
      User that you are looking for not exist
  </Typography>
);

export default NotFound;
