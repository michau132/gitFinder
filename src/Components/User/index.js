import React from 'react';
import PropTypes from 'prop-types';
import { Grid, withStyles } from '@material-ui/core';
import UserDetails from './UserDetails';
import UserAvatar from './UserAvatar';
import UserReposContainer from '../../Containers/UserReposContainer';
import UserReposHeader from './UserReposHeader';
import UserReposList from './UserReposList';

const styles = theme => ({
  root: {
    [theme.breakpoints.down('md')]: {
      flexDirection: 'column',
    },
    [theme.breakpoints.up('md')]: {
      flexDirection: 'row',
    },
  },
  avatarAndInfo: {
    [theme.breakpoints.down('md')]: {
      flexDirection: 'row',
    },
    [theme.breakpoints.up('md')]: {
      flexDirection: 'column',
    },
  },
});

const UserView = ({ user, classes }) => {
  const { userInfo, userRepos, ...restUserProps } = user;
  return (
    <Grid container spacing={40} className={classes.root}>
      <Grid
        item
        xs={12}
        sm={12}
        md={3}
        container
        justify="space-between"
        spacing={16}
        className={classes.avatarAndInfo}
      >
        <Grid item xs={12} sm={6} md="auto" container justify="center" alignItems="center">
          <UserAvatar avatarUrl={userInfo.avatar_url} />
        </Grid>
        <Grid item xs={12} sm={6} md="auto">
          <UserDetails userInfo={userInfo} />
        </Grid>
      </Grid>
      <UserReposContainer
        render={
          ({ hideSingleRepo, selectUserRepo, ...propsFromContainer }) => (
            <Grid item xs={12} sm={12} md={9} container justify="flex-end">
              <UserReposHeader {...restUserProps} {...propsFromContainer} />
              <UserReposList
                userRepos={userRepos}
                hideSingleRepo={hideSingleRepo}
                selectUserRepo={selectUserRepo}
              />
            </Grid>
          )}
      />
    </Grid>
  );
};

UserView.propTypes = {
  user: PropTypes.shape({
    userInfo: PropTypes.shape({
      login: PropTypes.string.isRequired,
      email: PropTypes.string,
      avatar_url: PropTypes.string,
      name: PropTypes.string,
    }).isRequired,
    userRepos: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        forks_count: PropTypes.number,
        stargazers_count: PropTypes.number,
        description: PropTypes.string,
        html_url: PropTypes.string.isRequired,
      }),
    ).isRequired,
  }).isRequired,
};

export default withStyles(styles)(UserView);
