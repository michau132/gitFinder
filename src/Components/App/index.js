import React from 'react';
import {
  HashRouter,
  Route,
  Switch,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import UserContainer from '../../Containers/UserContainer';
import styles from './style.css';
import HeaderContainer from '../../Containers/HeaderContainer';
import EmptyUser from '../EmptyUser';
import store from '../../utils/store';


const App = () => (
  <Provider store={store}>
    <HashRouter>
      <section className={styles.container}>
        <HeaderContainer />
        <Switch>
          <Route exact path="/" component={EmptyUser} />
          <Route path="/:user" component={UserContainer} />
        </Switch>
      </section>
    </HashRouter>
  </Provider>

);

export default App;
