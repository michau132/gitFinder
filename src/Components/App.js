import React from 'react';
import {
  HashRouter,
  Route,
  Switch,
} from 'react-router-dom';
import { configure } from 'mobx';
import { Paper, withStyles } from '@material-ui/core';
import styled, { createGlobalStyle } from 'styled-components';
import UserContainer from '../Containers/UserContainer';
import HeaderContainer from '../Containers/HeaderContainer';
import EmptyUser from './EmptyUser';
import Header from './Header';
import User from './User';
import NotFound from './NotFound';

configure({ enforceActions: 'observed' });

const styles = {
  paper: {
    backgroundColor: '#F5F5F5',
  },
};

const GlobalStyle = createGlobalStyle`
  html {
    box-sizing: border-box;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  body {
    background-color: #CCCCCC;
  }
  a {
    text-decoration: none;
  }
`;

const PaperWithPadding = styled(Paper)`
  padding: 8px;
`;

const App = ({ classes }) => (
  <HashRouter>
    <PaperWithPadding classes={{ root: classes.paper }}>
      <GlobalStyle />
      <PaperWithPadding>
        <HeaderContainer
          render={props => (
            <Header {...props} />
          )}
        />
        <Switch>
          <Route exact path="/" component={EmptyUser} />
          <Route
            exact
            path="/:user"
            render={() => (
              <UserContainer
                render={props => (
                  <User {...props} />
                )}
              />
            )}
          />
          <Route component={NotFound} />
        </Switch>
      </PaperWithPadding>
    </PaperWithPadding>
  </HashRouter>

);

export default withStyles(styles)(App);
