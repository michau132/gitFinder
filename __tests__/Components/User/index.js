import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { User } from '../../../src/Components/User';

describe('testing User component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <User />,
    );
  });
  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });
  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
