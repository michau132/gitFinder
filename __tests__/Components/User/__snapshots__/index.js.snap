// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing User component matching snapshot 1`] = `
<WithStyles(Grid)
  className="mediaQueries"
  container={true}
  spacing={40}
>
  <WithStyles(Grid)
    className="mediaQueries"
    container={true}
    item={true}
    justify="space-between"
    md={3}
    sm={12}
    spacing={16}
    xs={12}
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      md="auto"
      sm={6}
      xs={12}
    >
      <UserAvatar
        avatarUrl="https://via.placeholder.com/150"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md="auto"
      sm={6}
      xs={12}
    >
      <UserDetails
        informations={
          Object {
            "avatar_url": "https://via.placeholder.com/150",
            "email": "aoko123@gmail.com",
            "loctation": "Raszyn",
            "login": "aOko123",
            "name": "Adam Oko",
          }
        }
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    item={true}
    justify="flex-end"
    md={9}
    sm={12}
    xs={12}
  >
    <WithStyles(UserReposHeader) />
    <UserReposList
      handleHideSingleRepo={[MockFunction]}
      handleSelectUserRepo={[MockFunction]}
      repos={
        Array [
          Object {
            "description": "First repo description",
            "forks_count": 0,
            "html_url": "https://github.com/aOko123/firstRepo",
            "id": 1,
            "name": "firstRepo",
            "stargazers_count": 0,
          },
          Object {
            "description": "Second repo description",
            "forks_count": 0,
            "html_url": "https://github.com/aOko123/secondRepo",
            "id": 2,
            "name": "secondRepo",
            "stargazers_count": 0,
          },
        ]
      }
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
